doc/Language/regexes.pod6

```diff
@@ -1255,6 +1255,47 @@ list of predefined subrules is listed in
 L<S05-regex|https://design.perl6.org/S05.html#Predefined_Subrules> of design
 documents.
 
+=head1 X<Regex Interpolation|regex, Regex Interpolation>
+
+If you want to build a regex using pattern given at the runtime, regex
+interpolation is what you looking for.
+
+There are two ways you can interpolate a string into regex as a pattern,
+which are quite similar to interpolate it into a double-quote string.
+That is using C«<$pattern>» or C«<{$pattern.method}>».
+
+    my $text = 'camelia';
+    my $pattern = '\w+';
+    my $pattern_ = 'ailemac';
+    say $text ~~ / <$pattern> /;          # OUTPUT: «｢camelia｣␤»
+    say $text ~~ / <{$pattern}> /;        # OUTPUT: «｢camelia｣␤»
+    say $text ~~ / <{$pattern_.flip}> /;  # OUTPUT: «｢camelia｣␤»
+    # say $text ~~ / <$pattern_.flip> /;  !!Compile Error!!
+
+Note that this syntax causes L«implicit EVAL|
+/language/traps#<{$x}>_vs_$($x):_Implicit_EVAL», that is a known trap.
+
+When your pattern is a lexical string, there are two more syntax helping you
```

Comment:
Oh, I see, this is where you talk about just interpolating as a string. I would also mention that is the string to interpolate is statically typed as a Str or str (i.e., my Str $string = 'camelia'), it's much faster.

Reply:
```
my Str $pattern = '\d+';

say "42" ~~ /<$pattern>/;
``

Code above works too. So I think statically type `$pattern` as `Str` shows nothing.

Anyway, your other reviews are worthy. Thank you!

Reply:
Yeah, typing it doesn't change the behavior if you're using <$pattern>, but if you're just using $pattern it's *much* faster (~10x).

Reply:
got it. sorry for misunderstanding

Reply:
No worries, I'm glad we're adding some documentation about regex interpolation, it's a common question.